core/domain/voiceover_domain.py

```diff
@@ -0,0 +1,221 @@
+
+    def to_dict(self) -> EntityVoiceoversDict:
+        """Returns the dict representation of the EntityVoiceovers object.
+
+        Returns:
+            EntityVoiceoversDict. The dict representation of the
+            EntityVoiceovers object.
+        """
+        voiceovers_dict = {}
+        for content_id, voiceover_type_to_voiceover in self.voiceovers.items():
+            voiceovers_dict[content_id] = {
+                voiceover_type.value: voiceover_type_to_voiceover[
+                    voiceover_type].to_dict()
+                for voiceover_type in feconf.VoiceoverType
+            }
+        return {
+            'entity_id': self.entity_id,
+            'entity_type': self.entity_type,
+            'entity_version': self.entity_version,
+            'language_accent_code': self.language_accent_code,
+            'voiceovers': voiceovers_dict
+        }
+
+    @classmethod
+    def from_dict(
+        cls, entity_voiceovers_dict: EntityVoiceoversDict
+    ) -> EntityVoiceovers:
+        """Creates the EntityVoiceovers instance from the given dict.
+
+        Args:
+            entity_voiceovers_dict: EntityVoiceoversDict. The dict
+                representation of the EntityVoiceovers object.
+
+        Returns:
+            EntityVoiceovers. The EntityVoiceovers instance created using the
+            given dict.
+        """
+        content_id_to_voiceovers = {}
+        for content_id, voiceover_type_to_voiceover_dict in (
+                entity_voiceovers_dict['voiceovers'].items()):
+            content_id_to_voiceovers[content_id] = {
+                voiceover_type: state_domain.Voiceover.from_dict(
+                    voiceover_type_to_voiceover_dict[voiceover_type.value])
+                for voiceover_type in feconf.VoiceoverType
+            }
+        return cls(
+            entity_voiceovers_dict['entity_id'],
+            entity_voiceovers_dict['entity_type'],
+            entity_voiceovers_dict['entity_version'],
+            entity_voiceovers_dict['language_accent_code'],
+            content_id_to_voiceovers
+        )
+
+    def validate(self) -> None:
+        """Validates the EntityVoiceovers object."""
+        if not isinstance(self.entity_type, str):
+            raise utils.ValidationError(
+                'entity_type must be a string, received %s' % self.entity_type)
+        if not isinstance(self.entity_id, str):
+            raise utils.ValidationError(
+                'entity_id must be a string, received %s' % self.entity_id)
+        if not isinstance(self.entity_version, int):
+            raise utils.ValidationError(
+                'entity_version must be an int, received %s' %
+                self.entity_version)
+        if not isinstance(self.language_accent_code, str):
+            raise utils.ValidationError(
+                'language_accent_code must be a string, received %s' %
+                self.language_accent_code)
+        if not bool(re.match(
```

Comment:
Just wanted to ask if we can be sure of language code to be valid, I mean we have a bunch of valid language codes already defined, right? Will that be an issue if the language code is not one of them?

Reply:
Hi @lkbhitesh07 this validation is for language accent code i.e., language code with accent, examples: en-US, en-IN, hi-IN, etc.
Currently our codebase do not contains the exhaustive list of language accent codes (the list will be introduced in the upcoming PRs).
Hence we can't perform the validation as mentioned here.
Thank you.

Reply:
I see, thanks for the explanation.